redis/client.py

```diff
@@ -790,6 +790,39 @@ def client_kill(self, address):
         "Disconnects the client at ``address`` (ip:port)"
         return self.execute_command('CLIENT KILL', address)
 
+    def client_kill_filter(self, *filter_options):
+        """
+        Disconnects the client using a variety of filter options
+        :param filter_options: a tuple or list of filter options with
+        the following format:
+        (filter, value, filter, value,...) or
+        [filter, value, filter, value,...]
+        """
+        if not isinstance(filter_options, (list, tuple)) or not filter_options:
+            raise DataError("CLIENT KILL <filter> <value> ... ... <filter> "
+                            "<value> "
+                            "must be a non empty list or "
+                            "tuple to execute")
+        if len(filter_options) % 2 != 0:
+            raise DataError("CLIENT KILL <filter> <value> requires a filter "
+                            "and a value pair. Got %r" % (filter_options,))
+        filter_types = ('addr', 'id', 'type', 'skipme')
+        client_types = ('normal', 'master', 'slave', 'pubsub')
+        yes_no = ('yes', 'no')
+        for index in range(0, len(filter_options), 2):
+            key = str(filter_options[index]).lower()
+            value = str(filter_options[index + 1]).lower()
+            if key not in filter_types:
+                raise DataError("CLIENT KILL <filter> must be one of %r" % (
+                    filter_types,))
+            if key == 'type' and value not in client_types:
+                raise DataError("CLIENT KILL TYPE <value> "
+                                "must be one of %r" % (client_types,))
+            if key == 'skipme' and value not in yes_no:
+                raise DataError("CLIENT KILL SKIPME <value> "
+                                "must be one of %r" % (yes_no,))
+        return self.execute_command('CLIENT KILL', *filter_options)
```

Comment:
The filter types (addr, id, type, skipme) need to be sent as Tokens or bytes so that they don't get encoded. e.g., `Token.get_value('addr')`